Plutarch/Internal/Numeric.hs

```diff
@@ -149,7 +162,44 @@ instance Function Positive where
   {-# INLINEABLE function #-}
   function = functionMap @Integer coerce coerce
 
-{- | The addition operation.
+-- | @since WIP
+newtype PNatural (s :: S) = PNatural (Term s PInteger)
+  deriving stock
+    ( -- | @since WIP
+      Generic
+    )
+  deriving anyclass
+    ( -- | @since WIP
+      PlutusType
+    , -- | @since WIP
+      PIsData
+    , -- | @since WIP
+      PEq
+    , -- | @since WIP
+      POrd
+    )
+
+-- | @since WIP
+instance DerivePlutusType PNatural where
+  type DPTStrat _ = PlutusTypeNewtype
+
+-- | @since WIP
+instance PLiftable PNatural where
```

Comment:
Can we do what we do with PPositive instead here? I guess no cause ghc will be mad that there's no `Natural` constructor exposed right?
```haskell
deriving via
  DeriveNewtypePLiftable PNatural PInteger Natural
  instance
    PLiftable PNatural
```

Reply:
I tried that and it complained that `PInteger` doesn't lower to `Natural`.